client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -1,40 +1,106 @@
 import Masterbar from 'calypso/layout/masterbar/masterbar';
-import { getStatusForPlugin } from 'calypso/state/plugins/installed/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
-
+import { initiateThemeTransfer } from 'calypso/state/themes/actions';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
 /**
  * Style dependencies
  */
 import 'calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status/style.scss';
+import { getPurchaseFlowState } from 'calypso/state/plugins/marketplace/selectors';
 
 const StyledProgressBar = styled( ProgressBar )`
 	margin: 20px 0px;
 `;
 
-function WrappedMarketplacePluginSetup( {
-	pluginSlug = 'wordpress-seo',
-}: {
-	pluginSlug?: string;
-} ): JSX.Element {
+// Stages
+
+function WrappedMarketplacePluginSetup(): JSX.Element {
+	const translate = useTranslate();
+
+	const STAGE_1 = translate( 'Installing plugin' );
+	const STAGE_2 = translate( 'Activating plugin' );
+	const stages = [ STAGE_1, STAGE_2 ];
+
+	const [ currentStage, setCurrentStage ] = useState( STAGE_1 );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 0 );
+
+	const interval = 2000;
+
+	const dispatch = useDispatch();
 	const selectedSite = useSelector( getSelectedSite );
+	const transferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, selectedSite?.ID )
+	);
+	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+		getPurchaseFlowState
+	);
+
+	useEffect( () => {
+		if ( pluginSlugToBeInstalled ) {
+			dispatch( initiateThemeTransfer( selectedSite.ID, null, pluginSlugToBeInstalled ) );
+		} else if ( isPluginInstalledDuringPurchase ) {
+			alert( 'To be implemented: polling wait to be implemented to check for transfer' );
+		} else {
+			// Invalid State redirect to Yoast marketplace page for now, and maybe a marketplace home view in the future
+			page(
+				`/marketplace/product/details/wordpress-seo${
+					selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+				}`
+			);
+		}
+	}, [ dispatch, pluginSlugToBeInstalled ] );
+
+	useEffect( () => {
+		setTimeout(
+			() => {
+				if ( simulatedProgressPercentage < 95 ) {
+					setSimulatedProgressPercentage( ( percentage ) => percentage + 6 );
+				} else if ( simulatedProgressPercentage !== 100 && simulatedProgressPercentage !== 95 ) {
+					setSimulatedProgressPercentage( 95 );
+				}
+			},
+
+			interval
+		);
+	}, [ simulatedProgressPercentage ] );
+
+	useEffect( () => {
+		if ( transferStatus === 'complete' ) {
+			// TODO: Make sure the primary domain is set to the relevant domain before redireting to thank-you page
+			setSimulatedProgressPercentage( 100 );
+			setTimeout( () => {
+				page(
+					`/marketplace/thank-you${
+						selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+					}?flags=marketplace-yoast`
+				);
+			}, 4000 );
```

Comment:
It's not clear how we decided to wait for exactly 4 seconds here.

Reply:
Yes this is time was arrived at by few trial and error attempts to simulate the most reasonable amount of time it would take to do the transfer. Do you meant for me to add in a comment here? @tyxla ?

Reply:
This seems pretty fragile to me. Not only that if we change the transfer flow and it takes a different amount of time this may break, but it's also not obvious as to what we're doing here and why. Because of that, it may break in ways we can't predict. Is there any way we can make it more obvious? Perhaps put in a selector to check if the transfer happened already?

Reply:
Sorry I misspoke in the previous comment, This time was meant to allow a small window to give the customer an opportunity to see that the progress bar fully completed. The final fulfillment of the illusion 🤓 , will add it as a comment.

Reply:
I understand that but note that my comment about this being potentially fragile still stands.

Reply:
Yeah we are not on the same page :)
So this time out was meant to make sure the user see the final bit of the progress bar filling :).
I am already checking the transfer status with a selector  on the following line.
https://github.com/Automattic/wp-calypso/blob/640d4223234ec2ba2eb270a8c314484ca239ffcb/client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx#L84

Once the selector gives the green light I show the customer a completed progress bar. And then after a timeout of 4 seconds redirect to the thank you page.

Reply:
Thanks for confirming 😉  FWIW I understand and it will work well right now, but it can potentially be fragile as we iterate and data flow might change as we iterate in the future. The worst thing about that kind of bug is that they're hard to pre-empt, can be hard to reproduce, and can be hard to diagnose eventually. But feel free to ignore my warning, I don't want to block this PR 😉

Reply:
No worries I really enjoy discussing these potential pitfalls 👍 
Is it fragile though since the timeout is only set "after" we have the ok from the selector that the transfer has completed?
Do you have a better way to make the redirect wait after the transfer is complete to make the customer wait a bit longer in this page to show him a filled progress bar?

Reply:
It really depends on the use case, and I can't tell what the use case will be in the future. Right here and right now I don't see how it will break, so it is good to go. 

But in my work here (Jetpack Connect, authorization) I've seen instances where we relied on timeouts or component lifecycle being executed in the right order. While they were working well in the beginning, as we iterated and modified the code to do more things and go through more different flows, we silently introduced subtle bugs that we hadn't thought of before. But this might never get to be the case with this piece of code 😉  

What I'd suggest here is to keep in mind that time can cause problems, especially when we stack different pieces of logic on it, so the best thing to do is be explicit about what's going on. Perhaps we can add some inline documentation to explain it. Ideally, have some tests to cover the functionality and ensure it's still working as expected.